code/modules/reagents/reagents.dm

```diff
@@ -147,26 +149,27 @@
 	if(volume && holder)
 		remove_self(removed)
 
-/datum/reagent/proc/affect_blood(var/mob/living/carbon/M, var/alien, var/effect_multiplier)
-	return
+/datum/reagent/proc/affect_blood(mob/living/carbon/M, alien, effect_multiplier)
 
-/datum/reagent/proc/affect_ingest(var/mob/living/carbon/M, var/alien, var/effect_multiplier)
+/datum/reagent/proc/affect_ingest(mob/living/carbon/M, alien, effect_multiplier)
 	affect_blood(M, alien, effect_multiplier * 0.8)	// some of chemicals lost in digestive process
-	return
+	
+	var/mob/living/carbon/human/H = M
+	if(istype(H))
+		H.sanity.onReagent(src, effect_multiplier)
```

Comment:
These three lines of code repeats too much. it would be nice to make it a proc..